validator/sawtooth_validator/journal/publisher.py

```diff
@@ -357,23 +370,24 @@ def on_check_publish_block(self, force=False):
         """
         try:
             with self._lock:
-                if self._candidate_block is None and\
+                if self._chain_head is not None and\
+                        self._candidate_block is None and\
                         len(self._pending_batches) != 0:
                     self._candidate_block = self._build_block(self._chain_head)
-
                 if self._candidate_block and \
                         (force or len(self._pending_batches) != 0) and \
                         self._consensus.check_publish_block(self.
                                                             _candidate_block):
                     candidate = self._candidate_block
                     self._candidate_block = None
 
-                    self._finalize_block(candidate)
+                    if not self._finalize_block(candidate):
+                        return
+
                     # if no batches are in the block, do not send it out
                     if len(candidate.batches) == 0:
-                        LOGGER.info("No Valid batches added to block, " +
-                                    " dropping %s",
-                                    candidate.identifier[:8])
+                        LOGGER.info("No valid batches added to block, " +
+                                    " dropping %s", candidate)
```

Comment:
Will never reach this logging statement. If now blocks are added to a block (state_hash is None) you return False so this function is returned in line 385.